src/diagnostics.rs

```diff
@@ -613,11 +616,22 @@ impl Diagnostic {
 
     pub fn non_constant_case_condition(case: &str, range: SourceRange) -> Diagnostic {
         Diagnostic::SyntaxError {
-            message: format!("Non constant case condition: {}. Cannot evaluate for duplicates. Duplicate values may result in undefined runtime behaviour", case),
+            message: format!("{}. Cannot evaluate for duplicates. Duplicate values may result in undefined runtime behaviour", case),
```

Comment:
I would use this : 
"None constant variables are not supported in case conditions".
Not mentioning the duplicates here since the entire case is undefined, and not just the dublicates.